src/NuGet.Clients/NuGet.SolutionRestoreManager/VSNominationUtilities.cs

```diff
@@ -90,72 +91,65 @@ internal static RuntimeGraph GetRuntimeGraph(IEnumerable targetFrameworks)
         {
             var tfi = new TargetFrameworkInformation
             {
                 FrameworkName = GetTargetFramework(targetFrameworkInfo.Properties, projectFullPath),
                 TargetAlias = GetPropertyValueOrNull(targetFrameworkInfo.Properties, ProjectBuildProperties.TargetFramework)
             };
 
-            var ptf = MSBuildStringUtility.Split(GetPropertyValueOrNull(targetFrameworkInfo.Properties, ProjectBuildProperties.PackageTargetFallback))
-                                          .Select(NuGetFramework.Parse)
-                                          .ToList();
+            string? ptfString = GetPropertyValueOrNull(targetFrameworkInfo.Properties, ProjectBuildProperties.PackageTargetFallback);
+            List<NuGetFramework>? ptf = ptfString is not null
+                ? MSBuildStringUtility.Split(ptfString).Select(NuGetFramework.Parse).ToList()
+                : null;
 
-            var atf = MSBuildStringUtility.Split(GetPropertyValueOrNull(targetFrameworkInfo.Properties, ProjectBuildProperties.AssetTargetFallback))
-                                          .Select(NuGetFramework.Parse)
-                                          .ToList();
+            string? atfString = GetPropertyValueOrNull(targetFrameworkInfo.Properties, ProjectBuildProperties.AssetTargetFallback);
+            List<NuGetFramework>? atf = atfString is not null
+                ? MSBuildStringUtility.Split(atfString).Select(NuGetFramework.Parse).ToList()
+                : null;
 
             // Update TFI with fallback properties
             AssetTargetFallbackUtility.ApplyFramework(tfi, ptf, atf);
 
-
             tfi.RuntimeIdentifierGraphPath = GetPropertyValueOrNull(targetFrameworkInfo.Properties, ProjectBuildProperties.RuntimeIdentifierGraphPath);
 
-            if (targetFrameworkInfo.PackageReferences != null)
+            if (targetFrameworkInfo.Items is null)
             {
-                tfi.Dependencies.AddRange(
-                    targetFrameworkInfo.PackageReferences
-                        .Cast<IVsReferenceItem>()
-                        .Select(pr => ToPackageLibraryDependency(pr, cpvmEnabled)));
+                return tfi;
             }
 
-            if (targetFrameworkInfo is IVsTargetFrameworkInfo2 targetFrameworkInfo2)
+            if (targetFrameworkInfo.Items.TryGetValue(ProjectItems.PackageReference, out var packageReferences))
             {
-                if (targetFrameworkInfo2.PackageDownloads != null)
-                {
-                    tfi.DownloadDependencies.AddRange(
-                       targetFrameworkInfo2.PackageDownloads
-                           .Cast<IVsReferenceItem>()
-                           .SelectMany(ToPackageDownloadDependency));
-                }
+                tfi.Dependencies.AddRange(packageReferences.Select(pr => ToPackageLibraryDependency(pr, cpvmEnabled)));
```

Comment:
This should come in a future PR, but all these AddRange and Select in nomination codepaths can probably be rewritten. 

We probably just need an issue.

Reply:
* https://github.com/NuGet/Home/issues/13449